convert2rhel/pkghandler.py

```diff
@@ -90,6 +122,166 @@
+        # _RHSM_TMP_DIR do not contain any path components writable by
+        # a different user.
+        utils.mkdir_p(SUBMGR_RPMS_DIR)
+        utils.mkdir_p(_RHSM_TMP_DIR)
+
+        loggerinst.info("Downloading requested packages")
+        all_pkgs_to_install = self.pkgs_to_install + self.pkgs_to_update
+
+        if system_info.version.major == 7:
+            download_rhsm_pkgs(all_pkgs_to_install, _UBI_7_REPO_PATH, _UBI_7_REPO_CONTENT)
+        elif system_info.version.major == 8:
+            download_rhsm_pkgs(all_pkgs_to_install, _UBI_8_REPO_PATH, _UBI_8_REPO_CONTENT)
+
+        # installing the packages
+        rpms_to_install = [os.path.join(SUBMGR_RPMS_DIR, filename) for filename in os.listdir(SUBMGR_RPMS_DIR)]
+
+        loggerinst.info("Installing subscription-manager RPMs.")
+        loggerinst.debug("Rpms scheduled to be installed: %s" % utils.format_sequence_as_message(rpms_to_install))
+
+        _, ret_code = call_yum_cmd(
+            # We're using distro-sync as there might be various versions of the subscription-manager pkgs installed
+            # and we need these packages to be replaced with the provided RPMs from RHEL.
+            command="install",
+            args=rpms_to_install,
+            print_output=True,
+            # When installing subscription-manager packages, the RHEL repos are
+            # not available yet for getting dependencies so we need to use the repos that are
+            # available on the system
+            enable_repos=[],
+            disable_repos=[],
+            # When using the original system repos, we need YUM/DNF to expand the $releasever by itself
+            set_releasever=False,
+        )
+        if ret_code:
+            loggerinst.critical(
+                "Failed to install subscription-manager packages. See the above yum output for details."
+            )
+
+        installed_pkg_names = get_pkg_names_from_rpm_paths(rpms_to_install)
+        loggerinst.info(
+            "\nPackages we installed or updated:\n%s" % utils.format_sequence_as_message(installed_pkg_names)
+        )
+
+        # We could rely on these always being installed/updated when
+        # self.enabled is True but putting the values into separate attributes
+        # is more friendly if outside code needs to inspect the values.
+        # It is tempting to use the rpms we actually installed to populate these
+        # but we would have to extract both name and arch information from the
+        # rpms if we do that. (for the cornercases where a pkg for one arch is
+        # already installed and we have to install a different one.
+        self.installed_pkgs = self.pkgs_to_install[:]
+        self.updated_pkgs = self.pkgs_to_update[:]
+
+        super(RestorablePackageSet, self).enable()
+
+    def restore(self):
+        if not self.enabled:
+            return
+
+        loggerinst.info("Removing set of installed pkgs: %s" % utils.format_sequence_as_message(self.installed_pkgs))
```

Comment:
```suggestion
        logger.task("Convert: Remove installed RHSM packages")
        loggerinst.info("Removing set of installed pkgs: %s" % utils.format_sequence_as_message(self.installed_pkgs))
```

Reply:
This one will be problematic in the future as the class is intended to work for any set of packages.  I don't think we have any other package installs amongst the current actions, though, so addressing this when it comes up rather than now seems like a good plan.

Maybe we could deal with it by passing in the task header to print on rollback?